x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/Validation.java

```diff
@@ -175,18 +178,23 @@ public static final class Users {
          * @param username the username to validate
          * @param allowReserved whether or not to allow reserved user names
          * @param settings the settings which may contain information about reserved users
+         * @param maxLength the maximal allowed length a username can have
          * @return {@code null} if valid
          */
-        public static Error validateUsername(String username, boolean allowReserved, Settings settings) {
-            if (isValidUserOrRoleName(username) == false) {
-                return new Error(String.format(Locale.ROOT, INVALID_NAME_MESSAGE, "User"));
+        public static Error validateUsername(String username, boolean allowReserved, Settings settings, int maxLength) {
```

Comment:
This works. But I'd prefer hiding the `maxLength` parameter like how you did with the `FileRealmValidationUtil`. The length parameter is really an internal thing. We end up having 2 possible values due to mainly historical reasons, not because it is truly designed to accept multiple different length values. Therefore more specialised classes or methods seem to be a better fit instead of exposing the length as an argument for a public method. For an example, maybe the method can be private (or package private) and have a new public method `validateNativeUsername` which does not need the maxLength parameter. 

This likely implies more changes like relocating `FileRealmValidationUtil` so it can access the private (or package private) method. Since it's not critical, maybe we can let it pass in this PR and get back to it in future when we need to do something about roleMapping name validation (we don't have any today, but will likley need it). I'll let you make the call. Thanks!

Reply:
It's a valid point. I constrained myself (for no particular reason) not to make changes which affect native usages of this method. I'll adjust it since it will make the code cleaner and avoid misusage.